search_commands.go

```diff
@@ -0,0 +1,1215 @@
+package redis
+
+import (
+	"context"
+	"strings"
+)
+
+type SearchCmdable interface {
+	FT_List(ctx context.Context) *StringSliceCmd
+	FTAggregate(ctx context.Context, index string, query string) *MapStringInterfaceCmd
+	FTAggregateWithArgs(ctx context.Context, index string, query string, options *FTAggregateOptions) *MapStringInterfaceCmd
+	FTAliasAdd(ctx context.Context, index string, alias string) *StatusCmd
+	FTAliasDel(ctx context.Context, alias string) *StatusCmd
+	FTAliasUpdate(ctx context.Context, index string, alias string) *StatusCmd
+	FTAlter(ctx context.Context, index string, skipInitalScan bool, definition []interface{}) *StatusCmd
+	FTConfigGet(ctx context.Context, option string) *MapStringInterfaceCmd
+	FTConfigSet(ctx context.Context, option string, value interface{}) *StatusCmd
+	FTCreate(ctx context.Context, index string, options *FTCreateOptions, schema ...*SearchSchema) *StatusCmd
+	FTCursorDel(ctx context.Context, index string, cursorId int) *StatusCmd
+	FTCursorRead(ctx context.Context, index string, cursorId int, count int) *MapStringInterfaceCmd
+	FTDictAdd(ctx context.Context, dict string, term []interface{}) *IntCmd
+	FTDictDel(ctx context.Context, dict string, term []interface{}) *IntCmd
+	FTDictDump(ctx context.Context, dict string) *StringSliceCmd
+	FTDropIndex(ctx context.Context, index string) *StatusCmd
+	FTDropIndexWithArgs(ctx context.Context, index string, options *FTDropIndexOptions) *StatusCmd
+	FTExplain(ctx context.Context, index string, query string) *StringCmd
+	FTExplainWithArgs(ctx context.Context, index string, query string, options *FTExplainOptions) *StringCmd
+	FTInfo(ctx context.Context, index string) *MapStringInterfaceCmd
+	FTProfileSearch(ctx context.Context, index string, limited bool, query SearchQuery) *MapStringInterfaceCmd
+	FTProfileAggregate(ctx context.Context, index string, limited bool, query AggregateQuery) *MapStringInterfaceCmd
+	FTSpellCheck(ctx context.Context, index string, query string) *MapStringInterfaceCmd
+	FTSpellCheckWithArgs(ctx context.Context, index string, query string, options *FTSpellCheckOptions) *MapStringInterfaceCmd
+	FTSearch(ctx context.Context, index string, query string) *Cmd
+	FTSearchWithArgs(ctx context.Context, index string, query string, options *FTSearchOptions) *Cmd
+	FTSynDump(ctx context.Context, index string) *MapStringSliceInterfaceCmd
+	FTSynUpdate(ctx context.Context, index string, synGroupId interface{}, terms []interface{}) *StatusCmd
+	FTSynUpdateWithArgs(ctx context.Context, index string, synGroupId interface{}, options *FTSynUpdateOptions, terms []interface{}) *StatusCmd
+	FTTagVals(ctx context.Context, index string, field string) *StringSliceCmd
+}
+
+type FTCreateOptions struct {
```

Comment:
Given the variety, I think NewFTIndexWith<> functions are a good idea. You can talk me out of it for initial release, but I think it has value

Reply:
Agree. However, this PR is a low-level RediSearch implementation. NewFTIndexWith<> functions will be in a future update to simplify common use cases.